### PR TITLE
raftstore: set is_merging flag after restart

### PR DESCRIPTION
**Pull request by gengliqi**
cherry-pick #5868 to release-2.1

###  What have you changed?

Fix a region merge bug, set is_merging flag after restart.
When a crash/restart after prepareMerge being applied before compact log being applied, is_merging is not set so the compact log can not be ignored.
Then panic happens on getting entries for proposing commitMerge.
Add a failpoint test case, previous code panic almost every time I run this test.

###  What is the type of the changes?

- Bugfix (a change which fixes an issue)

###  How is the PR tested?

- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

File: tests/failpoints_cases/test_merge.rs [hunk 1]
```diff
@@ -421,3 +421,92 @@ fn test_node_merge_catch_up_logs_restart() {
...
+    for _ in 0..50 {
+        let state2 = cluster.truncated_state(left.get_id(), 1);
+        if state1.get_index() != state2.get_index() {
+            break;
+        }
+        sleep_ms(10);
+    }
+    // can schedule merge now
+    fail::remove(schedule_merge_fp);
+    //sleep_ms(100);
```
Comment (Connor1996) [inline tests/failpoints_cases/test_merge.rs]: clean it

**Comment by gengliqi** [comment]
I find this test is not stable in v2.1. I will fix it.

**Comment by AndreMouche** [comment]
friendly ping

**Comment by gengliqi** [comment]
I fix this unstable test. It also needs to do in master and 3.x. I will change them in the next merge-bug-fix pr. @BusyJay @Connor1996

**Pull request merged**